src/streamer.rs

```diff
@@ -246,26 +293,29 @@ fn retransmit(
 }
 
 /// Service to retransmit messages from the leader to layer 1 nodes.
-/// See `subscribers` for network layer definitions.
+/// See `crdt` for network layer definitions.
 /// # Arguments
 /// * `sock` - Socket to read from.  Read timeout is set to 1.
 /// * `exit` - Boolean to signal system exit.
-/// * `subs` - Shared Subscriber structure.  This structure needs to be updated and popualted by
-/// the accountant.
+/// * `crdt` - This structure needs to be updated and populated by the accountant and via gossip.
 /// * `recycler` - Blob recycler.
 /// * `r` - Receive channel for blobs to be retransmitted to all the layer 1 nodes.
 pub fn retransmitter(
     sock: UdpSocket,
     exit: Arc<AtomicBool>,
-    subs: Arc<RwLock<Subscribers>>,
+    crdt: Arc<RwLock<Crdt>>,
     recycler: BlobRecycler,
     r: BlobReceiver,
 ) -> JoinHandle<()> {
-    spawn(move || loop {
-        if exit.load(Ordering::Relaxed) {
-            break;
+    spawn(move || {
+        trace!("retransmitter started");
+        loop {
+            if exit.load(Ordering::Relaxed) {
+                break;
+            }
+            let _ = retransmit(&crdt, &recycler, &r, &sock);
```

Comment:
borrow-checker trouble when you return that error?

Reply:
no, we just drop that error because we only exit if `exit` is set

Reply:
needs a TODO, we should filter for expected errors and throw a general exception otherwise